jetty-util/src/main/java/org/eclipse/jetty/util/security/Credential.java

```
@@ -42,7 +44,7 @@
 {
     private static final long serialVersionUID = -7760551052768181572L;
     private static final Logger LOG = LoggerFactory.getLogger(Credential.class);
-    private static final ServiceLoader<CredentialProvider> CREDENTIAL_PROVIDER_LOADER = ServiceLoader.load(CredentialProvider.class);
+    private static final List<CredentialProvider> CREDENTIAL_PROVIDERS = TypeUtil.serviceStream(ServiceLoader.load(CredentialProvider.class)).collect(Collectors.toList());
 
```

Again behavior change, I think we should keep the iterator here, rather than ignoring the exceptions.

---

Here we were previously iterating through the ServiceLoader on every call to `getCredential`. I don't think we would want to keep that behaviour, but I will change it to throw on an error.